internal/repository/base.go

```diff
@@ -0,0 +1,511 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"sync"
+
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type database struct {
+	db *pgxpool.Pool
+}
+
+var (
+	pgError    error
+	pgInstance *database
+	pgOnce     sync.Once
+)
+
+func Connect(
+	ctx context.Context,
+	postgresEndpoint string,
+) (*database, error) {
+	pgOnce.Do(func() {
+		dbpool, err := pgxpool.New(ctx, postgresEndpoint)
+		if err != nil {
+			pgError = fmt.Errorf("unable to create connection pool: %w\n", err)
+		}
+
+		pgInstance = &database{dbpool}
+	})
+
+	return pgInstance, pgError
+}
+
+func (pg *database) Close() {
+	if pg != nil {
+		pg.db.Close()
+	}
+}
+
+func (pg *database) SetupDatabaseConfig(
```

Comment:
We should have a `GetConfig` method, so we can use these values at the components, and a `Config` model